Mlem/App/Views/Shared/Palette Components/PaletteDivider.swift

```diff
@@ -0,0 +1,20 @@
+//
+//  ColoredDivider.swift
+//  Mlem
+//
+//  Created by Eric Andrews on 2024-08-30.
+//
+
+import Foundation
+import SwiftUI
+
+/// Divider() that colors itself appropriately to the palette.
+/// DO NOT use in UIKit environments!
+struct PaletteDivider: View {
+    @Environment(Palette.self) var palette
+    
+    var body: some View {
+        Divider()
+            .overlay(palette.neutralAccent)
```

Comment:
In Dark mode this is way brighter than it was before. What about using a `Rectangle()` with a `palette.tertiary` fill?

Reply:
Nice catch! Just using `palette.tertiary` instead of `.neutralAccent` seems to do the trick, at least to my eye

Reply:
Hmm, it still looks a little too bright to me. Can we go darker again?

This PR:

![IMG_0868](https://github.com/user-attachments/assets/c7b7a0e0-63f7-49f1-b82b-84fade6e7028)

Before:

![Screenshot 2024-08-31 at 22 51 27](https://github.com/user-attachments/assets/a5a3e65d-8754-4de8-a745-4ed2e3fad06b)

Reply:
Since `.tertiary` is semi-transparent it'll be stacking ontop of the divider and producing a more opaque shade - adding `.opacity` or `.hidden` below the `.overlay` might fix that

Reply:
Yeah, that's a good call--Divider also changes between light and dark mode and is by default somewhat transparent. I've gotten a very nice approximation of system default--it shouldn't be noticeably different unless you're comparing screenshots at the pixel level, and even then it's pretty close.